_returners/splunk_nebula_return.py

```diff
@@ -302,11 +308,17 @@ def flushBatch(self):
 
         if len(self.batchEvents) > 0:
             headers = {'Authorization': 'Splunk ' + self.token}
-            try:
-                r = requests.post(self.server_uri, data=' '.join(self.batchEvents), headers=headers, verify=http_event_collector_SSL_verify, proxies=self.proxy, timeout=self.timeout)
-            except requests.exceptions.Timeout:
-                log.error('Request to splunk timed out. Not retrying.')
-            except Exception as e:
-                log.error('Request to splunk threw an error: {0}'.format(e))
+            self.server_uri = [x for x in self.server_uri if x[1] is not "bad"]
+            for server in self.server_uri:
+                try:
+                    r = requests.post(server[0], data=' '.join(self.batchEvents), headers=headers, verify=http_event_collector_SSL_verify, proxies=self.proxy, timeout=self.timeout)
+                    r.raise_for_status()
+                    server[1] = "good"
+                    break
+                except requests.exceptions.RequestException:
+                    log.error('Request to splunk server "%s" failed. Marking as bad.' % server[0])
```

Comment:
I'm thinking we should log at info, or maybe even debug. Because this log will be thrown every time the returner executes. Maybe add logic to throw an error log if *all* the endpoints get marked as bad.